### DISCUSSION TITLE
How do Irecord Grandorgue audio onto Fl studio?

### DISCUSSION THREAD

**Discussion by MrMayo246** [GrandOrgue/grandorgue#1817] [Help]
I've been trying to record audio from GO to FL and can't seem to figure out how. I've tried using my primary sound driver but FL won't pick up the audio for some reason. I inquired FL customer support about the issue and they just told me to use ASIO. I installed ASIO4all, but it doesn't appear as an audio output option in GO. Is there something I need to configure in ASIO4all or should I use a different method for running GO to FL? Any suggestions are greatly appreciated.

**Comment by oleg68** [comment]
1. First of all, setup your FL to work with your sound output. Try not to use ASIO4all, for example, Options->Audio Settings->Audio and select your device. If it does not have ASIO driver, select a directsound device. select 48000 as the Sample Rate
2. Then you need to download a GrandOrgue version with ASIO support. [This intermediate build](https://github.com/GrandOrgue/grandorgue/actions/runs/7971259824) is recommended
3. Run GrandOrgue, go to File->Settings->Audio, enable ``RtAudio: asio`` in ``Sound ports``, then select and a device in the ``Mapping Output``/``Audio Output`` and click to Change
4. Then Select ``RtAudio: asio: FL Studio ASIO and press to OK
5. Press to OK for exiting from GrandOrgue settings

**Reply by MrMayo246** [reply]
Which files do I download in the intermediate build? Just the windows.exe one?

**Reply by oleg68** [reply]
- For trying without installation use [grandorgue-3.14.0-0.20.windows.x86_64.zip](https://github.com/GrandOrgue/grandorgue/actions/runs/7971259824/artifacts/1258893398)
- If you like it and want to install, use [grandorgue-3.14.0-0.20.windows.x86_64.exe](https://github.com/GrandOrgue/grandorgue/actions/runs/7971259824/artifacts/1258893310)

**Reply by MrMayo246** [reply]
I tried installing the windows.exe version, but it closes a few seconds after every time I open it. Do you have any idea why?

**Reply by MrMayo246** [reply]
I decided to download the sourceforge version in the same location as the intermediate build files, and the software now works. However, when I select files, there is no option to select 'settings'. I did some digging around and I could not find 'RtAudio' either. Do you have any suggestions?

**Reply by oleg68** [reply]
The soureforge version is too old and does not contain some of features. It lacks ASIO support.

Use the Github version from https://github.com/GrandOrgue/grandorgue/releases

But the last one ([3.13.3-1]) has a bug #1811 that may cause a crash.

**Reply by MrMayo246** [reply]
I redownloaded v3.14, but it has the same issue as before when I was using v3.14. Seems that the crashing bug in v3.13 wasn't properly resolved in v3.14. Is there something I have to do on my end to resolve the issue?

**Reply by oleg68** [reply]
If GO crashes, please post here the crash dump and the exact version number.

See hee https://helgeklein.com/blog/creating-an-application-crash-dump/ how to enable it.

**Comment by MrMayo246** [comment]
[GrandOrgue.exe.6060.dmp](https://github.com/GrandOrgue/grandorgue/files/14468043/GrandOrgue.exe.6060.dmp)
(grandorgue-3.14.0-0.20.windows.x86_64.exe)

**Reply by oleg68** [reply]
The crash was introduced in #1660 if the config file has been created with a very old GrandOrgue version.

I've tried to fix it. Could you try [the new intermediate build](https://github.com/oleg68/GrandOrgue-official/actions/runs/8131877012)

I highly recommend to use the `.windows.x86_64.zip` instead of `.windows.x86_64.exe`.

If the crash still occurs, please post here the new crash dump.

**Reply by MrMayo246** [reply]
[GrandOrgue.exe.6924.dmp](https://github.com/GrandOrgue/grandorgue/files/14475426/GrandOrgue.exe.6924.dmp)
(grandorgue-3.14.0-0.23.windows.x86_64.zip)

**Comment by oleg68** [comment]
@MrMayo246 Seems GrandOrgue tries to use wrong `libGrandOrgueCore.dll` instead the one shipped with the zip archive. Possible, from an old version.

1. Could you try to find another file `libGrandOrgueCore.dll` in your system? If it exists, try to rename it and to repeat the test.
2. Could you post here your GrandOrgue config file (c:\Users\Cooldg\OneDrive\Application Data\GrandOrgueConfig).

**Reply by MrMayo246** [reply]
I found the .dll file and tried renaming it, but the GO application couldn't start because it "couldn't find libGrandOrgueCore.dll". As for the config file, what's strange is that I don't seem to have an Application Data folder in my OneDrive folder. I also tried searching for "GrandOrgueConfig" in my Users folder but only the GO help manual appeared.

**Reply by oleg68** [reply]
> I found the .dll file and tried renaming it, but the GO application couldn't start because it "couldn't find libGrandOrgueCore.dll"

Run GrandOrgue from a directory where you have unpacked the zip. This directory contains libGrandOrgueCore.dll. I need Grandorgue to use the .dll in this directory instead one from elsewhere.

> I don't seem to have an Application Data folder in my OneDrive folder

`Application Data` is a hidden directory in the user home directory. You have to enable displaying hidden files in your file explorer for finding it.

**Reply by MrMayo246** [reply]
[GrandOrgueConfig.zip](https://github.com/GrandOrgue/grandorgue/files/14489211/GrandOrgueConfig.zip)
This is my config file for GO. As for the .dll file, I'm not sure what you're instructing me to do but I tried deleting the other libGrandOrgueCore.dll file I had in my files to force GO to use the correct .dll file inside the original zip file, but GO still crashes.

**Reply by oleg68** [reply]
>  to force GO to use the correct .dll file inside the original zip file, 

Yes, exactly.

> but GO still crashes

Could you give me a crash dump from the correct .dll?

> This is my config file for GO

You posted the .lnk file to the config instead of the config file. Please give me the config file itself.
